### ISSUE TITLE
Stream output to memory buffer

### ISSUE THREAD

**Issue by MrMarble** [adrg/libvlc-go#108]
Is there a way to get the output stream?

I'd like to buffer some video for manual playing, currently I use `--sout #std{mux=ts,access=file,dst=-}` to output the data to the console and capture it manually to then use it as a buffer.

It would be better to access the stream directly, like with https://wiki.videolan.org/Documentation:Modules/smem/, but I'm not sure on how to do it (or if it's even possible)

**Comment by leroymusa** [comment]
Yes, you can use the `smem` module to capture the output stream directly into memory. Here's the approach:

```bash
vlc input_file --sout "#transcode{vcodec=RV24,acodec=s16l}:smem{video-prerender-callback=video_callback,audio-prerender-callback=audio_callback}"
```

You’ll need to implement the `video_callback` and `audio_callback` functions to handle the stream data. These callbacks give you direct access to  the video and audio buffers in memory for manual processing. This avoids writing to the console or a file.
